### ISSUE TITLE
enable uploading binary data via http

### ISSUE THREAD

**Issue by XertroV** [openplanet-nl/openplanet-issues#375]
atm, the only way to send data via POST request (or similar) is as a string. This is problematic for binary content since strings and `\x00` are not friends.
It'd be nice if a `MemoryBuffer` could be provided for POSTing (possibly requiring an appropriate content type header so that web servers are okay with it -- seems like that might be a thing)

My workaround atm is base64 encoding. Not sure if there's a better way atm.

**Comment by codecat** [comment]
My first thought would be to change `Body` from a `string` to a `MemoryBuffer@`, but that would probably break every single plugin that uses POST requests. API suggestions for this are welcome.

**Comment by XertroV** [comment]
maybe `Net::HttpRequest::SetBody(MemoryBuffer@ buf)`?
Then I can just replace `req.Body = b64encodedPayload` with `req.SetBody(buf)`.

For the most part, anyone sending proper binary probably won't need the `::HttpPost(url, data)` method (manually constructing the request is fine), but maybe we could add a `::HttpBinaryPost(url, MemoryBuffer@ data)` method too.

**Comment by codecat** [comment]
True. Then SetBody() could also accept a string, then we could slowly deprecated `req.Body` as a property.

**Comment by XertroV** [comment]
Oh yeah good point.

**Comment by codecat** [comment]
We should also support something like `SetBodyFile()` to pass a path so it can stream from disk.
